kernel/src/engine/arrow_expression/in_list.rs

```diff
@@ -0,0 +1,572 @@
+        // TODO: LargeListView - not fully supported by arrow yet
+        _ => {
+            return Err(Error::invalid_expression(format!(
+                "Expected right hand side to be list column, got: {:?}",
+                right_arr.data_type()
+            )));
+        }
+    };
+    let left_arr = evaluate_expression(left, batch, Some(&list_field.data_type().try_into()?))?;
+    let left_arr = if left_arr.data_type() == list_field.data_type() {
+        left_arr
+    } else {
+        cast(left_arr.as_ref(), list_field.data_type()).map_err(Error::generic_err)?
+    };
+    macro_rules! left_as {
+        ($t: ty ) => {
+            left_arr.as_primitive_opt::<$t>().ok_or_else(|| {
+                Error::invalid_expression(format!(
+                    "Cannot cast {} to {}",
+                    left_arr.data_type(),
+                    <$t>::DATA_TYPE
+                ))
+            })
+        };
+    }
+    Ok(match list_field.data_type() {
+        ArrowDataType::Utf8 | ArrowDataType::LargeUtf8
+            if right_arr.as_list_opt::<i32>().is_some() =>
+        {
+            eval_arrow_utf8(&left_arr, right_arr.as_list::<i32>())?
+        }
+        ArrowDataType::Int8 => eval_arrow(left_as!(Int8Type)?, &right_arr)?,
+        ArrowDataType::Int16 => eval_arrow(left_as!(Int16Type)?, &right_arr)?,
+        ArrowDataType::Int32 => eval_arrow(left_as!(Int32Type)?, &right_arr)?,
+        ArrowDataType::Int64 => eval_arrow(left_as!(Int64Type)?, &right_arr)?,
+        ArrowDataType::Float16 => eval_arrow(left_as!(Float16Type)?, &right_arr)?,
+        ArrowDataType::Float32 => eval_arrow(left_as!(Float32Type)?, &right_arr)?,
+        ArrowDataType::Float64 => eval_arrow(left_as!(Float64Type)?, &right_arr)?,
+        ArrowDataType::Timestamp(TimeUnit::Microsecond, _) => {
+            eval_arrow(left_as!(TimestampMicrosecondType)?, &right_arr)?
+        }
+        ArrowDataType::Date32 => eval_arrow(left_as!(Date32Type)?, &right_arr)?,
+        ArrowDataType::Decimal128(_, _) => eval_arrow(left_as!(Decimal128Type)?, &right_arr)?,
+        _ => {
+            return Err(Error::invalid_expression(format!(
+                "Cannot compare {} to {}",
+                left_arr.data_type(),
+                list_field.data_type()
+            )))
+        }
+    })
+}
+
+fn is_in_list(ad: &ArrayData, values: impl IntoIterator<Item = Option<Scalar>>) -> BooleanArray {
+    #[allow(deprecated)]
+    let inlist = ad.array_elements();
+    // `v IN (k1, ..., kN)` is logically equivalent to `v = k1 OR ... OR v = kN`, so evaluate
+    // it as such, ensuring correct handling of NULL inputs (including `Scalar::Null`).
+    values
+        .into_iter()
+        .map(|v| {
+            KernelPredicateEvaluatorDefaults::finish_eval_junction(
+                JunctionOperator::Or,
+                inlist
+                    .iter()
+                    .map(|k| Some(v.as_ref()?.partial_cmp(k)? == Ordering::Equal)),
```

Comment:
tiny readability nit:
```suggestion
                    .map(|k| Some(k.partial_cmp(v.as_ref()?)? == Ordering::Equal)),
```